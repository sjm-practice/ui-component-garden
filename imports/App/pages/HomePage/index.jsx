import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Button from "../../../components/base/Button";

const styles = {
  container: {
    display: "flex",
    height: "100vh",
    justifyContent: "center",
    alignItems: "center",
  },
  message: {
    color: "darkblue",
  },
};

const HomePage = ({ classes }) => (
  <div className={classes.container}>
    <Typography className={classes.message} variant="h4">
      HomePage.
    </Typography>
    <Button onClick={() => alert("clicked!")}>Push Me.</Button>
  </div>
);

HomePage.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(HomePage);
