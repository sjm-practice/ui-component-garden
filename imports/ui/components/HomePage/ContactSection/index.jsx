import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import ContactFormContainer from "./ContactFormContainer";

const styles = {
  container: {
    padding: "40px",
  },
  responsive: {
    maxWidth: "100%",
    height: "auto",
  },
};

const ContactSection = props => {
  const {
    classes: { container, responsive },
  } = props;

  return (
    <Grid container spacing={40} className={container}>
      <Grid item xs={12} md={6}>
        <img src="images/device.png" alt="Mobile Phone Device" className={responsive} />
      </Grid>
      <Grid item xs={12} md={6}>
        <ContactFormContainer />
      </Grid>
    </Grid>
  );
};

ContactSection.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(ContactSection);
