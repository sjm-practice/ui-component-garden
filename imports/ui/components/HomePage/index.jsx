import React from "react";
import Banner from "./Banner";
import ContactSection from "./ContactSection";

const HomePage = () => (
  <div>
    <Banner />
    <ContactSection />
  </div>
);

export default HomePage;
