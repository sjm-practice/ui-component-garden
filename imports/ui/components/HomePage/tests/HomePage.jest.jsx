/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import HomePage from "..";
import appReducer from "../../../../state/reducers";

describe("<HomePage />", () => {
  const testStore = createStore(appReducer);

  it("should match render snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={testStore}>
          <HomePage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
