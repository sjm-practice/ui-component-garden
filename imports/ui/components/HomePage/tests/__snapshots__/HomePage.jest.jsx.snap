// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match render snapshot 1`] = `
<div>
  <section
    className="banner-container"
  >
    <div
      className="banner-headline"
    >
      A platform to find, share, and discuss healthy places to shop and eat.
    </div>
    <a
      className="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-fab-19 MuiButton-extendedFab-20 MuiButton-contained-13 MuiButton-containedPrimary-14 MuiButton-raised-16 MuiButton-raisedPrimary-17 Banner-bannerButton-1"
      href="https://app.eatdrinkhealthy.co/"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noopener noreferrer"
      role="button"
      tabIndex="0"
      target="_blank"
    >
      <span
        className="MuiButton-label-3"
      >
        Web App
      </span>
      <span
        className="MuiTouchRipple-root-220"
      />
    </a>
  </section>
  <div
    className="MuiGrid-container-33 MuiGrid-spacing-xs-40-59 ContactSection-container-31"
  >
    <div
      className="MuiGrid-item-34 MuiGrid-grid-xs-12-73 MuiGrid-grid-md-6-95"
    >
      <img
        alt="Mobile Phone Device"
        className="ContactSection-responsive-32"
        src="images/device.png"
      />
    </div>
    <div
      className="MuiGrid-item-34 MuiGrid-grid-xs-12-73 MuiGrid-grid-md-6-95"
    >
      <div>
        <h4
          className="MuiTypography-root-131 MuiTypography-h4-146 MuiTypography-colorPrimary-161"
        >
          Contact Us
        </h4>
        <form
          id="contactForm"
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root-166 MuiFormControl-fullWidth-169"
            onDragStart={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "margin": "10px 0 10px 0",
              }
            }
          >
            <label
              className="MuiFormLabel-root-181 MuiInputLabel-root-170 MuiInputLabel-formControl-175 MuiInputLabel-animated-178 MuiInputLabel-outlined-180"
              data-shrink={false}
            >
              Name
            </label>
            <div
              className="MuiInputBase-root-201 MuiOutlinedInput-root-188 MuiInputBase-fullWidth-210 MuiInputBase-formControl-202"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-218 MuiOutlinedInput-notchedOutline-195"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-219"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input-211 MuiOutlinedInput-input-196"
                disabled={false}
                name="contactName"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root-166 MuiFormControl-fullWidth-169"
            onDragStart={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "margin": "10px 0 10px 0",
              }
            }
          >
            <label
              className="MuiFormLabel-root-181 MuiInputLabel-root-170 MuiInputLabel-formControl-175 MuiInputLabel-animated-178 MuiInputLabel-outlined-180"
              data-shrink={false}
            >
              Email
            </label>
            <div
              className="MuiInputBase-root-201 MuiOutlinedInput-root-188 MuiInputBase-fullWidth-210 MuiInputBase-formControl-202"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-218 MuiOutlinedInput-notchedOutline-195"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-219"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input-211 MuiOutlinedInput-input-196 MuiInputBase-inputType-214"
                disabled={false}
                name="contactEmail"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root-166 MuiFormControl-fullWidth-169"
            onDragStart={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "margin": "10px 0 10px 0",
              }
            }
          >
            <label
              className="MuiFormLabel-root-181 MuiInputLabel-root-170 MuiInputLabel-formControl-175 MuiInputLabel-animated-178 MuiInputLabel-outlined-180"
              data-shrink={false}
            >
              Message
            </label>
            <div
              className="MuiInputBase-root-201 MuiOutlinedInput-root-188 MuiInputBase-fullWidth-210 MuiInputBase-formControl-202 MuiInputBase-multiline-209 MuiOutlinedInput-multiline-194"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-218 MuiOutlinedInput-notchedOutline-195"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-219"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <textarea
                aria-invalid={false}
                className="MuiInputBase-input-211 MuiOutlinedInput-input-196 MuiInputBase-inputMultiline-213 MuiOutlinedInput-inputMultiline-198"
                disabled={false}
                name="contactMessage"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                rows={10}
                value=""
              />
            </div>
          </div>
          <button
            className="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-fab-19 MuiButton-extendedFab-20 MuiButton-contained-13 MuiButton-containedPrimary-14 MuiButton-raised-16 MuiButton-raisedPrimary-17 MuiButton-sizeLarge-26 ContactFormComponent-formButton-130"
            disabled={false}
            id="contactSubmit"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="submit"
          >
            <span
              className="MuiButton-label-3"
            >
              Send
            </span>
            <span
              className="MuiTouchRipple-root-220"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
